### PR TITLE
Set up collecting interpreter statistics

### PR DESCRIPTION
**Pull request by JWZ2018**
Task: #934 

Summary:
    - Set up interpreter statistics on the realm
    - Count object creations in ObjectValue constructor

 Test Plan:
    - Ran cli with logStatistics option and saw interpreterStats

### CODE REVIEW

**Comment by JWZ2018** [comment]
This PR only tracks object creation statistics. I will work on the other two statistics mentioned in the task progressively.

File: src/intrinsics/index.js [hunk 1]
```diff
@@ -460,5 +460,8 @@ export function initialize(i: Intrinsics, realm: Realm): Intrinsics {
   // 8.2.2, step 12
   AddRestrictedFunctionProperties(i.FunctionPrototype, realm);
 
+  //last step: let the realm know the intrinsics are initialized
```
Comment (NTillmann) [inline src/intrinsics/index.js]: Space after //

File: src/intrinsics/index.js [hunk 2]
```diff
@@ -460,5 +460,8 @@ export function initialize(i: Intrinsics, realm: Realm): Intrinsics {
...
+  realm.intrinsicsInitialized = true;
```
Comment (NTillmann) [inline src/intrinsics/index.js]: Looking at construct_realm.js from where this is called, it constructs a few more internal objects. So this assignment should go there at the end. And then the name `intrinsicsInitialized` is a bit off. Instead, I suggest to just reset the statistics counter, and make the increment in ObjectValue.js unconditional.

**Comment by hermanventer** [comment]
Is this something you are going to finish?

**Comment by JWZ2018** [comment]
I hope to get back to this sometime. If I do, I would be starting on a new PR. Closing this one.

**Pull request closed without merging**